File: 1732-100-1267-count-servers-that-communicate/1732-100-1267-count-servers-that-communicate.py
class Solution:
    def countServers(self, grid):
        n, m = len(grid), len(grid[0])
        row = [0] * n
        col = [0] * m
        cnt = 0

        for i in range(n):
            for j in range(m):
                if grid[i][j]:
                    row[i] += 1
                    col[j] += 1

        for i in range(n):
            for j in range(m):
                if grid[i][j] and (row[i] > 1 or col[j] > 1):
                    cnt += 1

        return cnt

File: 1732-100-1267-count-servers-that-communicate/1732-100-1267-count-servers-that-communicate.java
class Solution {
    public int countServers(int[][] grid) {
        int n = grid.length, m = grid[0].length;
        int[] row = new int[n];
        int[] col = new int[m];
        int cnt = 0;

        // Count servers in each row and column
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (grid[i][j] == 1) {
                    row[i]++;
                    col[j]++;
                }
            }
        }

        // Count servers that can communicate
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (grid[i][j] == 1 && (row[i] > 1 || col[j] > 1)) {
                    cnt++;
                }
            }
        }

        return cnt;
    }
}
